ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -55,60 +56,60 @@ where
         Self { u, v }
     }
 
-    pub fn compute_proof<λ: ArrayLength>(
-        &self,
-        r: F,
-    ) -> (ZeroKnowledgeProof<F, TwoNMinusOne<λ>>, ProofGenerator<F>)
+    pub fn compute_proof<λ: ArrayLength, I, J>(
+        u: I,
+        v: J,
+    ) -> ZeroKnowledgeProof<F, TwoNMinusOne<λ>>
     where
         λ: ArrayLength + Add + Sub<U1>,
         <λ as Add>::Output: Sub<U1>,
         <<λ as Add>::Output as Sub<U1>>::Output: ArrayLength,
         <λ as Sub<U1>>::Output: ArrayLength,
+        I: IntoIterator<Item = F>,
+        J: IntoIterator<Item = F>,
+        I::IntoIter: ExactSizeIterator,
+        J::IntoIter: ExactSizeIterator,
     {
-        debug_assert_eq!(self.u.len() % λ::USIZE, 0); // We should pad with zeroes eventually
+        let mut u = u.into_iter();
+        let mut v = v.into_iter();
+
+        debug_assert_eq!(u.len() % λ::USIZE, 0); // We should pad with zeroes eventually
```

Comment:
adding the support for non powers of lambda is actually pretty important, but also easy to do

Reply:
Yup. I think I may have already actually done it, now that I call `.unwrap_or(F::ZERO)`, but I check to make sure it's correct in a follow up diff when I remove these debug asserts.